src/app/app.component.ts

```diff
@@ -268,13 +348,74 @@ export class AppComponent implements AfterViewInit {
     this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
   }
 
+  async onToolbarOptionSelected(event) {
+    const customEvent = event as CustomEvent<ToolbarOption>;
+    const optionSelected: ToolbarOption = customEvent.detail;
+    await optionSelected.tag.handler();
+  }
+
+  openNewFolderDialog(): void {
+    this.dialog.open(NewFolderModalComponent, {
+      data: { makeNewFolder: this.makeNewFolder.bind(this) },
+    });
+  }
+
+  openEditColumnsDialog(): void {
+    this.dialog.open(EditColumnsModalComponent, {
+      data: { columnsSelected: this.selectedColumns, allColumnOptions: this.allPossibleColumns, updateColumns: this.updateColumns.bind(this),},
+    });
+  }
+
+  async makeNewFolder(folderName: string) {
+    if (folderName) {
+      if (!this.lfRepositoryBrowser?.nativeElement?.currentFolder) {
+        throw new Error('repositoryBrowser has no currently opened folder.');
+      }
+      await this.addNewFolderAsync(this.lfRepositoryBrowser?.nativeElement?.currentFolder, folderName);
+      await this.lfRepositoryBrowser?.nativeElement?.refreshAsync();
+    }
+    else {
+      throw new Error("didn't receive a folder name.");
+    }
+  }
+
+  async addNewFolderAsync(parentNode: LfTreeNode, folderName: string): Promise<void> {
+    if (!this.repoClient){
+      throw new Error('repoClient is undefined');
+    }
+    const entryId = (parentNode as LfRepoTreeNode).targetId ?? parseInt( parentNode.id, 10);
+
+    const requestParameters: { entryId: number; postEntryChildrenRequest: PostEntryChildrenRequest } = {
+      entryId,
+      postEntryChildrenRequest: new PostEntryChildrenRequest({
```

Comment:
assign as variable above for readability and easier stack trace (if it threw it would show as a separate line)